### PR TITLE
Lps 86236 fix2

### PR DESCRIPTION
**Pull request by arthurchan35**
Hi @natocesarrego 
From https://github.com/natocesarrego/liferay-portal/pull/131, if you dont remember.
Staging team has changed their mind that they wont depcreate staging in 7.2(https://issues.liferay.com/browse/LPS-86233), so I was approached by Victor Ware to restart this work.

### CODE REVIEW

**Comment by arthurchan35** [comment]
ci:test:sf

**Comment by arthurchan35** [comment]
ci:test:relevant

**Comment by arthurchan35** [comment]
Hi @natocesarrego 
Any update?

**Comment by natocesarrego** [comment]
Hi @arthurchan35,

First of all, sorry for the delay to answer you, but, as you may know ([Loop post](https://loop.liferay.com/home/-/loop/feed/17630566)), I was out of office from Monday to Wednesday of this week due to a conference, hence all my tasks are receiving delayed answers.

In regards to your fix, I've the following question to you:

Is there any other Portlet without admin part, like Calendar, that already addressed this issue? If yes, how it addressed this limitation? I'm asking this because, as you mentioned [here](https://github.com/natocesarrego/liferay-portal/pull/131#issuecomment-438888674), there isn't a real need of an admin part in Calendar Portlet and, at this moment, Calendar is under maintenance without any forecast of new features, which can difficult the addition of this admin part.

Thanks.

**Comment by natocesarrego** [comment]
Hi @arthurchan35,

I'm closing this pull due to inactivity. Once you have time to check my questions, please reopen the pull.

Best.

**Comment by arthurchan35** [comment]
ci:reopen

**Comment by arthurchan35** [comment]
Hi @natocesarrego ,
I am been stuck with some other tasks recently, sorry for the late reply. To your questions:
1. There are few portlets that are also mixing data site levels. Either instance level portlets have site level data handler or site level portlets have instance level data handler. However they are also having cause issue that produces portlet preference of 0, 3, 0
2. Yes, this fix itself does not address anything, but the fix(to fix 0, 3, 0) to its parent ticket will cause regression if we dont have this fix. And also like I mentioned before, it's conceptually wrong for a portlet that is able to import/export data not have a admin site level data handler.
3. I understand calendar is under maintanense right now, but I dont see why could not we merge this pr. For one, I dont think this fix will cause any problem. For two, even it causes any problem, I believe they can be easily fixed.

File: modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/portlet/CalendarAdminPortlet.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		"com.liferay.portlet.header-portlet-css=/css/main.css",
+		"com.liferay.portlet.icon=/icons/calendar.png",
+		"com.liferay.portlet.preferences-owned-by-group=true",
+		"com.liferay.portlet.preferences-unique-per-layout=false",
+		"javax.portlet.display-name=Calendar Admin",
+		"javax.portlet.expiration-cache=0",
+		"javax.portlet.name=" + CalendarPortletKeys.CALENDAR_ADMIN,
+		"javax.portlet.resource-bundle=content.Language",
+		"javax.portlet.security-role-ref=administrator",
+		"javax.portlet.supports.mime-type=text/html"
```
Comment (natocesarrego) [inline modules/apps/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/portlet/CalendarAdminPortlet.java]: Are all these properties useful for the Calendar Admin Portlet?

**Comment by natocesarrego** [comment]
Hi @arthurchan35,

Thanks for your answers! Let's ask the Calendar team's opinion about this fix. @inacionery, can you please take a look to this fix?

Best.

**Comment by arthurchan35** [comment]
@inacionery 
Can you help take a look at it?

**Comment by inacionery** [comment]
Hey @arthurchan35,

I could not understand what this PR is fixing, could you please fill in the issue with the **description**,  **steps to produce**, **actual result** and **expected result**.

Thanks!

**Comment by arthurchan35** [comment]
Hi @inacionery ,
Sorry for the confusion, but this PR is not about fixing a "current" bug. It's bit difficult to write down reproduce steps and results. Let me try to explain in this way:
So there is a bug LPS-86233 that publishing process will generate portlet preferences with ownerId = 0, ownerType = 3, plid = 0. See its description for detail.
I wrote a fix to correct LPS-86233, however that fix causes regression bugs. One regression bug comes from Calendar.
The reason Calendar causes regression is that Calendar portlet is not following the portlet data handler pattern that if a portlet exports data other than configurations/meta, this portlet needs a site level portlet data handler(or in another word, requires a admin portlet).
That's also why I said it's conceptually wrong for a portlet that exports/imports portlet data to be a instance level portlet.
There are also a few other portlets that are doing it wrong, so I have created several other sub tasks to address those.
You can refer to DL portlets for example, DL portlets have both admin and display portlets, the admin ones have site level data handler to do data exports/imports, while display portlets do not.


Thanks

**Comment by inacionery** [comment]
Hi @arthurchan35, 

The calendar is in maintenance mode. So no new features will be introduce and I understand this is a case in which the calendar was not designed to support. As you're saying is not a bug, this change should not be merged, sorry.

/cc @natocesarrego

**Comment by arthurchan35** [comment]
@inacionery ,
I understand Calendar is under maintenance but I think it's still necessary to merge this PR.
For one it's blocking my fix to LPS-86233, if you look at this in different angle: once LPS-86233 is fixed, it will cause a regression bug.
For two data handler issue in Calendar essentially is a design flaw that is a bug not being exposed.

**Comment by natocesarrego** [comment]
Closing due to inactivity.

**Pull request closed without merging**